File: Angular_Spring_SQL_Project/Front-End/src/app/services/admin/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { LocaldbService } from '../storage/localdb.service';

const BASIC_URL = "http://localhost:8000"

@Injectable({
  providedIn: 'root'
})

export class AdminService {

  constructor(private http: HttpClient) { }

  postCategory(CategoryDto: any) : Observable<any> {

    console.log("URL: " + BASIC_URL + "/admin/category");
    return this.http.post<[]>(BASIC_URL + "/admin/category" , CategoryDto, {
      headers: this.createAuthHeader()
    })
  }

  getAllCategories() : Observable<any> {

    console.log("URL: " + BASIC_URL + "/admin/categories");
    return this.http.get<[]>(BASIC_URL + "/admin/categories" , {
      headers: this.createAuthHeader()
    })
  }

  postProduct(categoryId: number, productDto: any) : Observable<any> {

    console.log("URL: " + BASIC_URL + "/admin/categories");

    return this.http.post<[]>(BASIC_URL + "/admin/" + categoryId + "/product", productDto,
      {
        headers: this.createAuthHeader()
      }
    )
  }

  getAllProductsByCategory(categoryId: number) : Observable<any> {

    console.log("URL: " + BASIC_URL + "/admin/" + categoryId + "/products");
    return this.http.get<[]>(BASIC_URL + "/admin/" + categoryId + "/products" , {
      headers: this.createAuthHeader()
    })
  }

  getProductById(productId: number) : Observable<any> {

    let url =  BASIC_URL + "/admin/product/" + productId
    console.log("URL", url)
    return this.http.get<[]>(url , {
      headers: this.createAuthHeader()
    })

  }

  updateProduct(productId: number, productDto: any) : Observable<any> {

    let url =  BASIC_URL + "/admin/product/" + productId

    console.log("URL", url)

    return this.http.put<[]>(url, productDto,
      {
        headers: this.createAuthHeader()
      }
    )
  }


  createAuthHeader(): HttpHeaders {
    const token = LocaldbService.getToken();
    console.log("Token", token)
    return new HttpHeaders({
      Authorization: `Bearer ${token}`
    });
  }

  deleteProductById(id: any) : Observable<any>  {

    let url =  BASIC_URL + "/admin/product/" + id
    console.log("URL: " , url);
    return this.http.delete<[]>(url , {
      headers: this.createAuthHeader()
    })

  }



}

File: Angular_Spring_SQL_Project/Back-End/src/main/java/org/example/simple_jwt_solution/services/interfaces/AdminService.java
package org.example.simple_jwt_solution.services.interfaces;

import java.io.IOException;
import java.util.List;

import org.example.simple_jwt_solution.dto.CategoryDto;
import org.example.simple_jwt_solution.dto.ProductDto;
import org.example.simple_jwt_solution.dto.ReservationDto;
import org.example.simple_jwt_solution.dto.ResultResponse;

public interface AdminService {

    ResultResponse postCategory(CategoryDto categoryDto) throws IOException;

    List<CategoryDto> getAllCategories();

    ResultResponse postProduct(Integer categoryId, ProductDto productDto) throws IOException;

    List<ProductDto> getAllProductsByCategory(Integer categoryId);

    ResultResponse deleteProduct(Integer productId);

    ProductDto getProductById(Integer productId);

    ResultResponse updateProduct(Integer productId, ProductDto productDto) throws IOException;

    List<ReservationDto> getAllReservations();

    ResultResponse changeReservationStatus(Integer reservationId, String status);
}
